### DISCUSSION TITLE
How to make default non-automatic requests invalid and re-issue requests

### DISCUSSION THREAD

**Discussion by AaronConlon** [TanStack/query#7296] [Q&A] [answered]
Hi everyone, I want to ask if I set the enabled to false when configuring the options for useQuery, how can I resolve the issue of being unable to trigger a refetch when invalidating the query? Thanks!

**Comment by TkDodo** [comment, marked as answer]
the only thing that bypasses `enabled:false` is the `refetch` function returned from `useQuery`.

**Reply by AaronConlon** [reply]
It still feels weird to me that if I have a query that I won't request immediately, and I want to re-request it from anywhere via queryKey , but I can only do this via the refetch function returned when I define it, which seems cumbersome to pass around between components. Although I can think of ways to achieve this, such as saving refetch via context , state management libraries, or even global variables, in order to let me use the refetch function elsewhere 😂

**Reply by TkDodo** [reply]
You usually don't want enabled:false + refetch - you want a variable condition in enabled, as described in the lazy queries section of the docs:

https://tanstack.com/query/latest/docs/framework/react/guides/disabling-queries#lazy-queries

**Reply by AaronConlon** [reply]
Got it, thank you
